File: week2/MarshalLee/BestTimeToBuyAndSellStock.cs
using System;

public class BestTimeToBuyAndSellStock
{
    public int MaxProfit(int[] prices)
    {
        int minPrice = prices[0];
        int maxProfit = 0;

        for (int i = 1; i < prices.Length; i++)
        {
            minPrice = Math.Min(minPrice, prices[i]);
            maxProfit = Math.Max(maxProfit, prices[i] - minPrice);
        }

        return maxProfit;
    }
}

File: week2/esadmsk_w2/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        max_profit = 0
        min_price = float('inf')
        for price in prices:
            min_price = min(min_price, price)
            max_profit = max(max_profit, price - min_price)
        return max_profit
